File: dotnet/Lang/String.cs
namespace Otterop.Lang;
public class String
{
    private string Wrapped;

    private String(string wrapped)
    {
        this.Wrapped = wrapped;
    }

    public static String Wrap(string wrapped)
    {
        return new String(wrapped);
    }

    public int CompareTo(String other)
    {
        if (other == null) return -1;
        return this.Wrapped.CompareTo(other.Wrapped);
    }

    public override string ToString()
    {
        return Wrapped;
    }
}

File: ts/lang/string.ts
export class String {
    private wrapped : string;

    private constructor(wrapped: string) {
        this.wrapped = wrapped;
    }

    public static wrap(wrapped: string) : String {
        return new String(wrapped);
    }

    public compareTo(other: String) : number {
        if (!other) return -1;
        if (this.wrapped < other.wrapped) return -1;
        else if (this.wrapped > other.wrapped) return 1;
        else return 0;
    }

    toString() : string {
        return this.wrapped;
    }
}
